src/RequestCookieCollection.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Cookies;
+
+use InvalidArgumentException;
+
+/**
+ * A CookieCollection helps to work with many cookies at once and to read request cookies.
+ *
+ * @see Cookie
+ *
+ */
+final class RequestCookieCollection
+{
+    /**
+     * @var Cookie[] The cookies in this collection (indexed by the cookie name).
+     *
+     * @psalm-var array<string, Cookie>
+     */
+    private array $cookies = [];
+
+    /**
+     * CookieCollection constructor.
+     *
+     * @param array|Cookie[] $cookies The cookies that this collection initially contains.
+     */
+    public function __construct(array $cookies = [])
+    {
+        foreach ($cookies as $cookie) {
+            if (!($cookie instanceof Cookie)) {
```

Comment:
Suggest use array instead of `Cookie` object: `array<string (cookie name), string (cookie value)>`

Reply:
done